### PR TITLE
dnn(tflite): add support for MAXIMUM layer

### LINKED ISSUE #26433: dnn Tflite module SQUARED_DIFFERENCE, SUM, MAXIMUM, SQRT operands support
### Describe the feature and motivation

Opencv 4.10.

Hello!
The team and I wrote a neural network on Tensorflow-lite, which uses the operands SQUARED_DIFFERENCE, SUM, MAXIMUM, SQRT. We use it through the dnn opencv module, and we encountered the problem that these operands are not implemented.
Any help with implemeting them?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ramukhsuya**
Fixes #26433
This PR adds support for the `MAXIMUM` layer in the TFLite importer.
It maps the TFLite `MAXIMUM` opcode to the existing OpenCV Element-wise `Max` operation.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
@dkurt Could you take a look?

File: modules/dnn/src/tflite/tflite_importer.cpp [hunk 1]
```diff
@@ -577,7 +578,10 @@ void TFLiteImporter::parseEltwise(const Operator& op, const std::string& opcode,
     }
     else if (opcode == "SQRT" && !isOpInt8) {
         layerParams.type = "Sqrt";
-    } else {
+    }
+    else if (opcode == "MAXIMUM" && !isOpInt8) {
+    layerParams.set("operation", "max");
```
Comment (asmorkalov) [inline modules/dnn/src/tflite/tflite_importer.cpp]: Please fix formatting.

File: modules/dnn/src/tflite/tflite_importer.cpp [hunk 2]
```diff
@@ -288,6 +288,7 @@ TFLiteImporter::DispatchMap TFLiteImporter::buildDispatchMap()
     dispatch["ADD"] = dispatch["MUL"] = dispatch["SUB"] =
         dispatch["SQRT"] = dispatch["DIV"] = dispatch["NEG"] =
         dispatch["RSQRT"] = dispatch["SQUARED_DIFFERENCE"] = &TFLiteImporter::parseEltwise;
+        dispatch["MAXIMUM"] = &TFLiteImporter::parseEltwise;
```
Comment (dkurt) [inline modules/dnn/src/tflite/tflite_importer.cpp:291]: Fix alignment

**Comment by asmorkalov** [comment]
@ramukhsuya Could you create simple unit test for the enabled layer. You need to create small model with 1-2-3 layers (including Maxinum layer), submit it with reference input and output (generated with TFLite) to opencv_extra with the same branch name and add test code to https://github.com/opencv/opencv/blob/4.x/modules/dnn/test/test_tflite_importer.cpp

File: modules/dnn/src/tflite/tflite_importer.cpp (hunk 1 above)
Comment (ramukhsuya) [reply modules/dnn/src/tflite/tflite_importer.cpp]: Fixed the spacing and indentation as requested

File: modules/dnn/src/tflite/tflite_importer.cpp (hunk 2 above)
Comment (ramukhsuya) [reply modules/dnn/src/tflite/tflite_importer.cpp:291]: Fixed alignment

**Comment by ramukhsuya** [comment]
Added a unit test for the layer and the TFLite model and .npy reference input and output files are uploaded to opencv_extra with the same branch name as requested

File: modules/dnn/test/test_tflite_importer.cpp [hunk 3]
```diff
@@ -283,6 +283,30 @@ TEST_P(Test_TFLite, face_blendshapes)
...
+
+    net.setPreferableBackend(backend);
+    net.setPreferableTarget(target);
+
+    Mat input_x = blobFromNPY(
+        findDataFile("dnn/tflite/maximum_input_x.npy"));
+    Mat input_y = blobFromNPY(
+        findDataFile("dnn/tflite/maximum_input_y.npy"));
+
+    net.setInput(input_x, "input_x");
```
Comment (dkurt) [inline modules/dnn/test/test_tflite_importer.cpp]:
```
[ RUN      ] Test_TFLite.maximum/0, where GetParam() = OCV/CPU
unknown file: Failure
C++ exception with description "OpenCV(4.13.0-dev) /build/precommit_linux64/4.x/opencv/modules/dnn/src/net_impl.cpp:1406: error: (-204:Requested object was not found) Requested blob "input_x" not found in function 'setInput'
" thrown in the test body.
```

Comment (ramukhsuya) [reply modules/dnn/test/test_tflite_importer.cpp]: Thanks for the feedback . I am debugging the TFLite error (input indexing and build configration). Will update with clean fix soon .
so while i work on it i have converted the PR to draft

Comment (dkurt) [reply modules/dnn/test/test_tflite_importer.cpp]: @ramukhsuya you may try this script for test generation: https://github.com/opencv/opencv_extra/blob/4.x/testdata/dnn/tflite/generate.py

File: modules/dnn/test/test_tflite_importer.cpp [hunk 4]
```diff
@@ -283,6 +283,25 @@ TEST_P(Test_TFLite, face_blendshapes)
...
+    Mat input_x = blobFromNPY(findDataFile("dnn/tflite/maximum_input_x.npy"));
+    Mat input_y = blobFromNPY(findDataFile("dnn/tflite/maximum_input_y.npy"));
+
+    net.setInput(input_x, "x");
+    net.setInput(input_y, "y");
+
+    Mat out = net.forward();
+
+    Mat ref = blobFromNPY(findDataFile("dnn/tflite/maximum_output.npy"));
+    normAssert(ref, out);
```
Comment (dkurt) [inline modules/dnn/test/test_tflite_importer.cpp]: Please use a higher threshold for FP16 target. CUDA_FP16 test failed:
```
[ RUN      ] Test_TFLite.maximum/1, where GetParam() = CUDA/CUDA_FP16
/home/ci/opencv/modules/dnn/test/test_common.impl.hpp:84: Failure
Expected: (normL1) <= (l1), actual: 0.000195324 vs 1e-05
  |ref| = 5.5
/home/ci/opencv/modules/dnn/test/test_common.impl.hpp:87: Failure
Expected: (normInf) <= (lInf), actual: 0.000781298 vs 0.0001
  |ref| = 5.5
[  FAILED  ] Test_TFLite.maximum/1, where GetParam() = CUDA/CUDA_FP16 (4 ms)
```

Comment (ramukhsuya) [reply modules/dnn/test/test_tflite_importer.cpp]: I’ve updated the threshold for FP16 targets (CUDA/OpenCL): both L1 and L∞ thresholds are now set to 1e-3 to account for FP16 precision.

**Comment by asmorkalov** [comment]
@dkurt Please take a look again. Test passed now.
